app/models/blueprint.rb

```diff
@@ -1,4 +1,6 @@
 class Blueprint < ApplicationRecord
+  has_many :versions, :class_name => 'Blueprint',
+           :foreign_key => :original_blueprint_id, :primary_key => :original_blueprint_id
```

Comment:
@Fryguy is there a parent / children relationship? are you saying that `original_blueprint_id` is essentially a `parent_id` (or basically a `root_id`)? So there is essentially a hierarchy here?

I see this more as versioning, without the hierarchy, and essentially the `original_blueprint_id` is the just the document_id.

----

`original_blueprint_id` could be left as nil to represent that it is the root document, or it's own id. I've seen it done as both.

Having `def root? ; original_blueprint_id == id ; end` ends up being much simpler than `def root? ; original_blueprint_id.nil? ; end` - especially in the sql case. I think it makes up for the double write.